dxr/plugins/clang/menus.py

```diff
@@ -205,10 +205,15 @@ def from_condensed(cls, tree, prop):
         # We ignore the def location clang gives us since it can be wrong, but
         # its existence means this ref isn't itself a def.
         search_for_def = prop.get('defloc') is not None
-        return cls(tree,
-                   (search_for_def, prop['qualname'],
-                    'has_overriddens' in prop, 'has_overrides' in prop),
-                   qualname=prop['qualname'])
+        ret = cls(tree,
+                  (search_for_def, prop['qualname'],
+                   'has_overriddens' in prop, 'has_overrides' in prop),
+                  qualname=prop['qualname'])
+        if not search_for_def:
+            # Prefer FunctionRefs with definitions over FunctionRefs without
+            # definitions.
+            ret.sort_order += 0.1
```

Comment:
A nice quick fix. We just need to keep in mind (or in a comment somewhere conspicuous) that we're essentially reserving the tenths place for this purpose, lest somebody else start adding or subtracting also and nullify this intermittently.